frontend/src/scenes/insights/insightSceneLogic.tsx

```diff
@@ -233,6 +236,18 @@ export const insightSceneLogic = kea<insightSceneLogicType>([
                     : null
             },
         ],
+        maxContext: [
+            (s) => [s.insight, s.insightLogicRef],
+            (insight: Partial<QueryBasedInsightModel>, insightLogicRef: any): MaxContextSelector => {
+                // Only provide insight context when in view mode
+                // In edit mode, MaxAI has the `create_and_query_insight` tool available
+                const isInViewMode = insightLogicRef?.logic.values.isInViewMode
+                if (!insight || !insight.short_id || !insight.query || !isInViewMode) {
+                    return []
+                }
+                return [createMaxContextHelpers.insight(insight)]
```

Comment:
This genuinely confused me when I was running the app and trying out context in insights – I was like "where has my insight gone". I know it's just ported behavior, but thinking it'd be clearer to not consider edit/view mode here

(Separately: Feeling like the "Edit insight" tool should be available in view mode too – it's un-magical that I have to manually click "Edit" myself to get going)